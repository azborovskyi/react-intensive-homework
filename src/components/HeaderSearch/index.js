import React from 'react';
import { func } from 'prop-types';

const HeaderSearch = (props) => {

    const _onFilterChange = ({ target: { value }}) => {
        const { filterTextChange } = props;

        filterTextChange(value);
    };

    return (
        <header>
            <h1>Планировщик задач</h1>
            <input
                onChange = { _onFilterChange }
            />
        </header>
    );
};

HeaderSearch.propTypes = {
    filterTextChange: func
};

export default HeaderSearch;
