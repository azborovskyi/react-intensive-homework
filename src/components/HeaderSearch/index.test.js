// Core
import React from 'react';
import HeaderSearch from '.';
import dom from 'react-test-renderer';

const renderTree = dom.create(<HeaderSearch filterTextChange = { () => {} } />).toJSON();

test('HeaderSearch should correspond to its snapshot', () => {
    expect(renderTree).toMatchSnapshot();
});