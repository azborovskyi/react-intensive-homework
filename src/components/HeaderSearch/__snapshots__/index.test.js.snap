// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSearch should correspond to its snapshot 1`] = `
<header>
  <h1>
    Планировщик задач
  </h1>
  <input
    onChange={[Function]}
  />
</header>
`;
